### PR TITLE
ci: add lychee link checker

### PR DESCRIPTION
**Pull request by supratikmahato**
### ❓ Type of change

- [ ] 📖 Documentation (updates to the documentation, readme or JSdoc annotations)
- [ ] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [ ] 👌 Enhancement (improving an existing functionality like performance)
- [x] ✨ New feature (a non-breaking change that adds functionality)
- [x] 🧹 Chore (updates to the build process or auxiliary tools and libraries)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

This pull request adds Lychee, a Rust-based link checker that checks if the links are valid or not.

Note that the `GITHUB_TOKEN` is **only** required to increase the rate limit when checking GitHub links and does not serve any special purpose.

### 📝 Checklist

- [ ] I have linked an issue or discussion.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by TechGenius7777** [comment]
I messed up with the commit message / pull request title.

What should I do now?

**Comment by Barbapapazes** [comment]
> I messed up with the commit message / pull request title.
> 
> What should I do now?

Hello, you can simply update your PR title since PR are squashed (so commit won't be used)!

**Comment by TechGenius7777** [comment]
> > I messed up with the commit message / pull request title.
> > What should I do now?
> 
> Hello, you can simply update your PR title since PR are squashed (so commit won't be used)!

Thank you so much!

**Review by danielroe — CHANGES_REQUESTED** [review]
I don't seem to have permission to push, so would you be able to address the broken links? 🙏

**Comment by TechGenius7777** [comment]
So exactly what I have to do?

Do I need to fix the broken links?

**Comment by TechGenius7777** [comment]
Pull request is ready to be merged!

File: .github/workflows/check-links.yml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+  push:
+  pull_request:
+
+jobs:
+  linkChecker:
+    runs-on: ubuntu-latest
+    steps:
+      # Cache lychee results (e.g. to avoid hitting rate limits)
+      - name: Restore lychee cache
+        uses: actions/cache@v3
```
Comment (danielroe) [inline .github/workflows/check-links.yml:13]:
```suggestion
        uses: actions/cache@704facf57e6136b1bc63b828d79edcd491f0ee84 # v3.3.2
```

File: .github/workflows/check-links.yml [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+        with:
+          path: .lycheecache
+          key: cache-lychee-${{ github.sha }}
+          restore-keys: cache-lychee-
+
+      # check links with Lychee
+      - uses: actions/checkout@v3
+
+      - name: Lychee link checker
+        uses: lycheeverse/lychee-action@v1
```
Comment (danielroe) [inline .github/workflows/check-links.yml:23]:
```suggestion
        uses: lycheeverse/lychee-action@ec7614d7605b47efb08dc370f6d0a71884cba944 # for v1.8.0
```

File: .github/workflows/check-links.yml [hunk 3]
```diff
@@ -0,0 +1,37 @@
+name: Check links with Lychee
+
+on:
+  push:
+  pull_request:
+
```
Comment (danielroe) [inline .github/workflows/check-links.yml:6]:
```suggestion

# Remove default permissions of GITHUB_TOKEN for security
# https://docs.github.com/en/actions/using-jobs/assigning-permissions-to-jobs
permissions: {}

```

File: .github/workflows/check-links.yml [hunk 4]
```diff
@@ -0,0 +1,37 @@
+name: Check links with Lychee
+
+on:
+  push:
+  pull_request:
```
Comment (danielroe) [inline .github/workflows/check-links.yml:5]:
```suggestion
  push:
    paths:
      - "docs/**"
      - "*.md"
    branches:
      - main
  pull_request:
    paths:
      - "docs/**"
      - "*.md"
    branches:
      - main
```

**Review by danielroe — APPROVED** [review]

**Comment by TechGenius7777** [comment]
Thank you!

**Pull request merged**